app/default/routes.py

```diff
@@ -239,16 +251,35 @@ def continue_application(application_id):
 
 
 @default_bp.route("/submit_application", methods=["POST"])
+@login_required
 def submit_application():
     application_id = request.form.get("application_id")
+    current_user = g.account_id
+    application = get_application_data(application_id, as_dict=True)
+    application_owner = application.account_id
+    if current_user != application_owner:
+        abort(401, f"User {current_user} attempted to submit_application for application {application_id}, owned by {application_owner}")
+    submitted = format_payload_and_submit_application(application_id)
+
+    response_weeks = 8
+
+    return render_template(
+        "application_submitted.html",
+        application_id=submitted.get("id"),
+        application_reference=submitted.get("reference"),
+        application_email=submitted.get("email"),
+        response_weeks=response_weeks,
+    )
+
+def format_payload_and_submit_application(application_id):
     payload = {"application_id": application_id}
     submission_response = requests.post(
         Config.SUBMIT_APPLICATION_ENDPOINT.format(
             application_id=application_id
         ),
         json=payload,
     )
-    submitted = submission_response.json()
+    submitted = submit_application(application_id)
```

Comment:
I haven't quite understood this line - does the submit_application() route call format_payload_and_submit_application() which then calls submit_application() again?

Reply:
No, great spot, thanks. Line 282 isn't meant to be there - error on my part during the rebase.